### PR TITLE
NewURI event for unhash

### PR DESCRIPTION
**Pull request by DancingAxolotl**
Note: this PR relies on changes in #49 and `enable-uns-support` is set as base branch to make keeping track of changes easier.

### CODE REVIEW

File: src/main/java/com/unstoppabledomains/resolution/contracts/uns/Registry.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    private static final String namingServiceName = "CNS";
+
+    public Registry(String url, String address, IProvider provider) {
+        super(namingServiceName, url, address, provider);
+    }
+
+    public String getDomainName(BigInteger tokenID) {
+        try {
+            String[] args = { Utilities.tokenIDToNamehash(tokenID) };
+            List<Tuple> logs = fetchLogs("earliest", "NewURI", args);
```
Comment (DeRain) [inline src/main/java/com/unstoppabledomains/resolution/contracts/uns/Registry.java:23]: @DancingAxolotl Good idea. If Infura allow to use `earliest` as a block number we should use it in other libraries and avoid using deploymentBlock from config.

**Review by DeRain — APPROVED** [review]
👍 Please make sure we have tests for newly introduced code

File: src/main/java/com/unstoppabledomains/resolution/contracts/uns/Registry.java (hunk 1 above)
Comment (DancingAxolotl) [reply src/main/java/com/unstoppabledomains/resolution/contracts/uns/Registry.java:23]: It works, but I'm not sure if it is using the deployment block or just the earliest block in the network.

**Comment by DancingAxolotl** [comment]
> 👍 Please make sure we have tests for newly introduced code

That's a good point. The tests that we have currently, are closer to end-to-end integration tests.
The new code is covered by them but on the high level.

Do you think I should add more specific unit tests for the new code?

**Comment by DeRain** [comment]
> > 👍 Please make sure we have tests for newly introduced code
> 
> That's a good point. The tests that we have currently, are closer to end-to-end integration tests.
> The new code is covered by them but on the high level.
> 
> Do you think I should add more specific unit tests for the new code?

I think few unit tests for BaseContract and other utility methods will be fine

**Pull request merged**